crates/biome_formatter/src/builders.rs

```diff
@@ -610,6 +610,35 @@ impl<Context> Format<Context> for Space {
     }
 }
 
+/// Optionally inserts a single space if the given condition is true.
+///
+/// # Examples
+///
+/// ```
+/// use biome_formatter::format;
+/// use biome_formatter::prelude::*;
+///
+/// # fn main() -> FormatResult<()> {
+/// // the tab must be encoded as \\t to not literally print a tab character ("Hello{tab}World" vs "Hello\tWorld")
+/// let elements = format!(SimpleFormatContext::default(), [text("a"), conditional_space(true), text("b")])?;
+///
+/// assert_eq!("a b", elements.print()?.as_code());
+///
+/// let nospace = format!(SimpleFormatContext::default(), [text("a"), conditional_space(false), text("b")])?;
+///
+/// assert_eq!("ab", elements.print()?.as_code());
+/// # Ok(())
+/// # }
+/// ```
+#[inline]
+pub fn conditional_space(should_insert: bool) -> Option<Space> {
+    if should_insert {
+        Some(Space)
+    } else {
+        None
+    }
+}
+
```

Comment:
I added this `conditional_space` function just for clarity at the callsites. Not sure if it has too much utility outside of this option, but I like the "branchless" flow from the caller side while still making it clear that the space may or may not exist.

Reply:
Instead of using `conditional_space(cond)` at call site, did you try `cond.then_some(space())`?

Reply:
> Instead of using `conditional_space(cond)` at call site, did you try `cond.then_some(space())`?

That wouldn't be very ergonomic when writing the IR, and it saves us a bunch of branching.

I can totally see the usage of this builder because our infra can understand `Option`. I would call this function `maybe_space` (sorry for the bike-shedding) because it's a prefix we already used through the code base,.

Reply:
I can definitely agree with `maybe_*`. Honestly idk why I didn't go with that in the first place since it's normally what I end up using lol.

I didn't actually know `.then_some` existed when I wrote this code (still very new to Rust), and had I known I might've used that instead. But I do like the declarative nature of saying `maybe_` for things, along with the advantage of working with any boolean value, not just an `Option`.

PR should be updated with the name change now.